File: typegate/tests/runtimes/temporal/temporal.py
from typegraph import t, typegraph, Policy, Graph
from typegraph.providers.temporal import TemporalRuntime


@typegraph()
def temporal(g: Graph):
    public = Policy.public()
    temporal = TemporalRuntime("<name>", "<host>")
    arg = t.struct({"some_field": t.string()})

    g.expose(
        public,
        start=temporal.start_workflow("<workflow_type>", arg),
        query=temporal.query_workflow("<query_type>", arg),
        signal=temporal.signal_workflow("<signal_name>", arg).reduce(
            {"workflow_id": "1234"}
        ),
        describe=temporal.describe_workflow(),
    )

File: typegate/tests/runtimes/temporal/temporal.ts
// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
// SPDX-License-Identifier: Elastic-2.0

import { Policy, t, typegraph } from "@typegraph/sdk/mod.js";
import { TemporalRuntime } from "@typegraph/sdk/providers/temporal.js";

typegraph("temporal", (g: any) => {
  const pub = Policy.public();
  const temporal = new TemporalRuntime("<name>", "<host>");
  const arg = t.struct({ some_field: t.string() });

  g.expose(
    {
      start: temporal.startWorkflow("<workflow_type>", arg).withPolicy(pub),
      query: temporal.queryWorkflow("<query_type>", arg).withPolicy(pub),
      signal: temporal.signalWorkflow("<signal_name>", arg).reduce({
        workflow_id: "1234",
      }).withPolicy(pub),
      describe: temporal.describeWorkflow().withPolicy(pub),
    },
  );
});
